### PR TITLE
feat: added the host detail view

### PR DESCRIPTION
**Pull request by rahulkeswani101**
### Summary

- Added the Host Detailed view which will open on click of the host list records. 
- Added tabs and initial details for hosts detailed view.

#### Related Issues / PR's

https://github.com/SigNoz/engineering-pod/issues/1878

#### Screenshots

image

#### Affected Areas and Manually Tested Areas

- Tested the initial flow and on click of record the proper details are populated.
- Data is consistent on the drawer and closing the drawer is functioning as expected. 
- Host name can be empty so, I need to handle this scenario while filtering the data. I will handle this in a separate PR. 

----

> [!IMPORTANT]
> Add `HostMetricDetail` component for detailed host metrics view with integrated styles and functionality in the host list.
> 
>   - **New Feature**:
>     - Added `HostMetricDetail` component in `index.tsx` for detailed host metrics view.
>     - Introduced `HostDetailProps` type in `HostMetricDetail.interfaces.ts` for component props.
>     - Added `VIEWS` enum and `VIEW_TYPES` constant in `constants.ts` for view management.
>   - **Styles**:
>     - Created `HostMetricsDetail.styles.scss` for styling the host detail drawer and its elements.
>     - Updated `InfraMonitoring.styles.scss` to include styles for clickable rows.
>   - **Integration**:
>     - Integrated `HostMetricDetail` in `HostsList.tsx` to open on row click and close on drawer close.
>     - Modified `utils.tsx` to rename `ioWait` to `wait` in `HostRowData` and update related logic.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=SigNoz%2Fsignoz&utm_source=github&utm_medium=referral)<sup> for e2db119273a921352352a13499d17f1c3c8f44c5. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by ahmadshaheer** [comment]
A few observations, @rahulkeswani101

image

1. the status is not visible
2. can we change the UI of status and operating system to badge? 
image

3. the CPU and Memory usage progress bars are not reflecting the percentage
4. Can we only show the tabs that have data? / alternatively, can we show empty state?

**Comment by ahmadshaheer** [comment]
> A few observations, @rahulkeswani101
> 
> image
> 
> 1. the status is not visible
> 2. can we change the UI of status and operating system to badge?
>    image
> 3. the CPU and Memory usage progress bars are not reflecting the percentage
> 4. Can we only show the tabs that have data? / alternatively, can we show empty state?

@rahulkeswani101 

Please ignore the 1, 2, and 3 points. I just noticed that they are fixed in https://github.com/SigNoz/signoz/pull/6297

File: frontend/src/components/HostMetricsDetail/index.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	X,
+} from 'lucide-react';
+import { useState } from 'react';
+
+import { VIEW_TYPES, VIEWS } from './constants';
+import { HostDetailProps } from './HostMetricDetail.interfaces';
+
+// import Metrics from './Metrics';
+// import Overview from './Overview';
+
```
Comment (ahmadshaheer) [inline frontend/src/components/HostMetricsDetail/index.tsx:22]: can you remove the commented code?

**Review by ahmadshaheer — APPROVED** [review]

File: frontend/src/components/HostMetricsDetail/index.tsx (hunk 1 above)
Comment (rahulkeswani101) [reply frontend/src/components/HostMetricsDetail/index.tsx:22]: @ahmadshaheer this has been fixed in other PR.

File: frontend/src/components/HostMetricsDetail/index.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+							<div className="view-title">
+								<DraftingCompass size={14} />
+								Traces
+							</div>
+						</Radio.Button>
+					</Radio.Group>
+
+					{/* {selectedView === VIEW_TYPES.OVERVIEW && <Overview host={host} />} */}
+					{/* {selectedView === VIEW_TYPES.METRICS && <Metrics host={host} />} */}
+				</>
```
Comment (SagarRajput-7) [inline frontend/src/components/HostMetricsDetail/index.tsx:148]: commented code

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/components/HostMetricsDetail/index.tsx (hunk 2 above)
Comment (rahulkeswani101) [reply frontend/src/components/HostMetricsDetail/index.tsx:148]: @SagarRajput-7 This has been fixed in other PR.

File: frontend/src/components/HostMetricsDetail/index.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+											host.cpu > 0.8 ? 'var(--error-500)' : 'var(--success-500)'
+										}
+									/>
+								</div>
+								<div className="progress-wrapper">
+									<Progress
+										percent={Math.round(host.memory * 100)}
+										size="small"
+										strokeColor={
+											host.memory > 0.8 ? 'var(--error-500)' : 'var(--success-500)'
```
[the remark this answers is not in the document]
Comment (rahulkeswani101) [reply frontend/src/components/HostMetricsDetail/index.tsx:75]: @SagarRajput-7 this has been fixed in other PR.

**Pull request merged**